logviewer/logviewer-lib.pl

```diff
@@ -0,0 +1,133 @@
+# logviewer-lib.pl
+# Functions for the syslog module
+
+BEGIN { push(@INC, ".."); };
+use WebminCore;
+&init_config();
+%access = &get_module_acl();
+
+# can_edit_log(&log|file)
+# Returns 1 if some log can be viewed/edited, 0 if not
+sub can_edit_log
+{
+return 1 if (!$access{'logs'});
+local @files = split(/\s+/, $access{'logs'});
+local $lf;
+if (ref($_[0])) {
+	$lf = $_[0]->{'file'} || $_[0]->{'pipe'} || $_[0]->{'host'} ||
+	      $_[0]->{'socket'} || $_[0]->{'cmd'} ||
+	      ($_[0]->{'all'} ? "*" : "users");
+	}
+else {
+	$lf = $_[0];
+	}
+foreach $f (@files) {
+	return 1 if ($f eq $lf || &is_under_directory($f, $lf));
+	}
+return 0;
+}
+
+# 
+# Returns standard 
+sub get_systemctl_cmds
+{
+my $lines = $config{'lines'} || 1000;
+return !&has_command('journalctl') ? () : (
+	{ 'cmd' => "journalctl --lines $lines -p 0..3",
```

Comment:
What do the numbers 0 to 3 mean?

Reply:
Priority.

```
-p, --priority=
           Filter output by message priorities or priority ranges. Takes
           either a single numeric or textual log level (i.e. between
           0/"emerg" and 7/"debug"), or a range of numeric/text log
           levels in the form FROM..TO. The log levels are the usual
           syslog log levels as documented in [syslog(3)](https://www.man7.org/linux/man-pages/man3/syslog.3.html), i.e.
           "emerg" (0), "alert" (1), "crit" (2), "err" (3),
           "warning" (4), "notice" (5), "info" (6), "debug" (7). If a
           single log level is specified, all messages with this log
           level or a lower (hence more important) log level are shown.
           If a range is specified, all messages within the range are
           shown, including both the start and the end value of the
           range. This will add "PRIORITY=" matches for the specified
           priorities.
```

I believe the text version would be more self-documenting. e.g. `emerg` instead of `0`.